test/end-to-end-tests/single-root-ctest/project-folder/CMakePresets.json

```diff
@@ -0,0 +1,44 @@
+{
+  "version": 2,
+  "configurePresets": [
+    {
+      "name": "2Successes",
+      "description": "Sets generator, build and install directory, vcpkg",
+      "generator": "Ninja",
+      "binaryDir": "${workspaceFolder}/build",
+      "cacheVariables": {
```

Comment:
We need to ensure that we define the CMAKE_C_COMPILER and CMAKE_CXX_COMPILER, depending on what OS this is run on.

Reply:
@gcampbell-msft i added hidden configure preset that hold compiler specifications for linux and windows systems. These presets are then mixed up with those setting up the tests to have all cases covered.